.github/workflows/main.yml

```diff
@@ -1,42 +1,188 @@
-name: Deploy GH Pages
-
-on:
-  push:
-    branches: 
-      - master 
-    paths:
-      - 'docs/*.md'
-      - 'docs/quickstart'
-      - '*.md'
-
-jobs:
-  deploy-gh-pages:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout master
-        uses: actions/checkout@v2
-        with:
-          path: master
-
-      - name: Checkout gh-pages branch
-        uses: actions/checkout@v2
-        with:
-          ref: 'gh-pages'
-          path: gh-pages
-    
-      - name: Copy site files
-        run: |
-          cp master/docs/*.md gh-pages/
-          cp -r master/docs/quickstart gh-pages/_pages
-          cp master/CODE_OF_CONDUCT.md master/CONTRIBUTING.md gh-pages/_pages/
-
-      - name: Commit GH Pages
-        run: |
-          cd gh-pages
-          git config user.name armada-admin
-          git config user.email admin@armadaproject.io
-          git add . _pages
-          git diff --quiet && git diff --staged --quiet || git commit -m "Updating Github Pages branch with latest Master changes"
-          git push
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+# Pre-merge TODO list
+# 1. change DOCKERTOKEN and DOCKERUSERNAME
+name: CI
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  GO111MODULE: "on"
+  GOPATH: "/home/runner/work/armada/armada/go"
+  GOCACHE: "/home/runner/.cache/go-build"
+
+on: 
+  push:
+    branches:
+      - '**'
+    pull_request:
+      branches:
+        - master
+    tags-ignore:
+      - '**'
+    paths-ignore:
+      - 'docs/*.md'
+      - 'docs/quickstart'
+      - '*.md'
+      - '**/release.yml'
+      - '**/pages.yml'
+
+jobs:
+  code_styles:
+    name: Code style
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'

```

Comment:
`go.mod` defined `1.12` but here we use `1.16`, would check with devs if this can be bumped to `1.16`, if yes update `go.mod` as well
_same for test job_

Reply:
I've checked the current state, it's not defined exact version...so I must did this on my own 🤣 
As you said, I would leave this to devs/team to define as we can only advice best-practice (which you just did 🙏 )
thanks @76creates 🚀